worker-jobtracking/src/main/java/com/hpe/caf/worker/jobtracking/JobTrackingWorkerReporter.java

```diff
@@ -89,12 +89,40 @@ public JobTrackingWorkerReporter() throws JobReportingException {
      */
     @Override
     public void reportJobTaskProgress(final String jobTaskId, final int estimatedPercentageCompleted) throws JobReportingException {
-        try (Connection conn = getConnection()) {
-            //TODO - FUTURE: pass estimatedPercentageCompleted to the database function
-            report(conn, jobTaskId, JobStatus.Active);
-        } catch (SQLException se) {
-            throw new JobReportingException(MessageFormat.format("Failed to report the progress of job task {0}. {1}", jobTaskId, se.getMessage()), se);
+
+        int retryCount = 0;
+        final int maxRetries = 5;
```

Comment:
A combination of a couple of things I think. The use of a long JobId (e.g. '7cf26494bc00400_1B2XGydNQ5K_wv1XLsWmW') and many nested sub-tasks results in Postgres truncating the index name. This combined with additional set of Job Tracking Workers caused the previous retry count of 1 to be exceeded. I have no control over when the key violation occurs in the first place as it all depends on what each transaction is doing at that time. I have only managed to repro this once since it happened but a 2nd retry happen to resolve it. I have simply bumped this up in case it occurs again. If you can't think of an alternative solution given there is no predictability when this will happen, then let me know.